lib/puppet/type/archive.rb

```diff
@@ -103,7 +103,7 @@ def should_to_s(value)
   newparam(:source) do
     desc 'archive file source, supports http|https|ftp|file uri.'
     validate do |value|
-      unless value =~ URI.regexp(%w(http https file ftp))
+      unless value =~ URI.regexp(%w(http https file ftp)) || value =~ %r{^([A-Z]:|\/)}
```

Comment:
In my opinion, trusting the mothership to do the right thing is a good idea until we specifically find out that we need to work around it.